homeassistant/components/modbus/modbus.py

```diff
@@ -322,11 +329,7 @@ async def async_setup(self) -> bool:
             func = getattr(self._client, entry.func_name)
             self._pb_call[entry.call_type] = RunEntry(entry.attr, func)
 
-        async with self._lock:
-            if not await self.hass.async_add_executor_job(self._pymodbus_connect):
-                err = f"{self.name} connect failed, retry in pymodbus"
-                self._log_error(err, error_state=False)
-                return False
+        self.hass.async_create_task(self.async_pymodbus_connect())
```

Comment:
All tasks created with `async_create_task` are still awaited to finish the startup phase. To create a task that is not awaited, use:
```suggestion
        self.hass.async_create_background_task(
            self.async_pymodbus_connect(), “modbus-connect”
        )
```